src/cms/views/events/event_actions.py

```diff
@@ -60,3 +64,32 @@ def delete(request, event_id, region_slug, language_code):
         'region_slug': region_slug,
         'language_code': language_code,
     })
+
+
+@login_required
+@region_permission_required
+def search_poi_ajax(request):
+    data = json.loads(request.body.decode('utf-8'))
+    poi_query = data.get('query_string')
+    region_slug = data.get('region_slug')
+
+    logger.info('Ajax call: Live search for POIs with query "%s"', poi_query)
+
+    region = Region.objects.get(slug=region_slug)
+
+    poi_query_result = []
+
+    if poi_query:
+        # Only query for pois from current region
+        for poi in region.pois.all():
+            for language in poi.languages:
+                poi_translation = poi.get_public_translation(language_code=language.code)
+                if poi_translation is not None and poi_query.lower() in poi_translation.title.lower():
+                    poi_query_result.append(poi)
+                    break
```

Comment:
This could speed up the execution a bit (especially when there are many pois, the difference could be noticeable)...
I'm aware of the fact that with my solution we cannot keep track of the versions and might include results where only the title of an old version matches, but I'm not sure if this is a problem...

The field lookup selector `icontains` is the case-insensitive variant of `contains`.
```suggestion
    poi_query_result = region.pois.prefetch_related('translations').filter(
        archived=False,
        translations__title__icontains=poi_query,
        translations__status=status.PUBLIC
    ).distinct()
```

Reply:
Oh, if you go with my solution, you have to include
```
from ...constants import status
```
in the header of course.

Reply:
I had very similar code for the query in the beginning but the problem was exactly what you mentioned above - old versions occurred in the results. This is especially confusing if we have an older version of a POI with a similar name than another POI in the current version (e.g. POI 1, version 1: "test location", version 2: "new location"; POI 2, version 1: "another test"). In the example, the query "test" would not yield both POIs, which could lead to confusion.

I therefore deliberately chose to implement it the way it currently is. I think that is a point we should discuss about maybe in the call tomorrow.

Reply:
Ok. In some cases, this might even be helpful (e.g. if a location is renamed and a user only remembers the old name), but especially when locations are created as draft with temporary titles, the results can be misleading.

Somehow I have the feeling that we are able to achieve the desired queries with the advanced Django expressions [Q()](https://docs.djangoproject.com/en/3.0/topics/db/queries/#complex-lookups-with-q-objects) and [F()](https://docs.djangoproject.com/en/3.0/ref/models/expressions/#f-expressions) :sweat_smile:
Maybe I'll have a look into it...

Reply:
I've never heard of those options before but they actually sound pretty promising!

Reply:
Ok, now I think I got it :sweat_smile: 
No need for Q&F, a subquery was the way to go:

```suggestion
    # All latest revisions of a POI (one for each language)
    latest_public_poi_revisions = POITranslation.objects.filter(
        poi=OuterRef('pk'),
        status=status.PUBLIC
    ).order_by('language', '-version').distinct('language').values('id')
    # All POIs which are not archived and have a latest public revision which contains the query
    poi_query_result = region.pois.prefetch_related('translations').filter(
        archived=False,
        translations__in=Subquery(latest_public_poi_revisions),
        translations__title__icontains=poi_query
    ).distinct()
```

Reply:
(Additionally needs:
```
from django.db.models import OuterRef, Subquery

from ...constants import status

from ...models import Event, Region, POITranslation
```